pywinauto/hooks.py

```diff
@@ -0,0 +1,290 @@
+                """Execute when keyboard low level event was catched"""
+                try:
+                    key_code = 0xFFFFFFFF & kb_data_ptr[0]
+                    current_key = self.ID_TO_KEY[key_code]
+                    event_type = self.event_types[0xFFFFFFFF & event_code]
+
+                    if event_type == 'key down':
+                        self.pressed_keys.append(current_key)
+
+                    if event_type == 'key up':
+                        self.pressed_keys.remove(current_key)
+
+                    event = KeyboardEvent(current_key, event_type, self.pressed_keys)
+
+                    if self.handler != 0:
+                        self.handler(event)
+
+                finally:
+                    return windll.user32.CallNextHookEx(self.keyboard_id, code, event_code, kb_data_ptr)
+
+            keyboard_pointer = _callback_pointer(keyboard_low_level_handler)
+
+            windll.kernel32.GetModuleHandleA.restype = wintypes.HMODULE
+            windll.kernel32.GetModuleHandleA.argtypes = [wintypes.LPCWSTR]
+            self.keyboard_id = windll.user32.SetWindowsHookExA(self.WH_KEYBOARD_LL, keyboard_pointer,
+                                                               windll.kernel32.GetModuleHandleA(None),
+                                                               0)
+
+        if self.mouse_hook:
+            def mouse_low_level_handler(code, event_code, kb_data_ptr):
+                """Execute when mouse low level event was catched"""
+                try:
+                    current_key = self.MOUSE_ID_TO_KEY[event_code]
+                    if current_key != 'Move':
+                        event_type = self.MOUSE_ID_TO_EVENT_TYPE[event_code]
+                        event = MouseEvent(current_key, event_type)
+
+                        if self.handler != 0:
+                            self.handler(event)
+
+                finally:
+                    return windll.user32.CallNextHookEx(self.mouse_id, code, event_code, kb_data_ptr)
+
+            mouse_pointer = _callback_pointer(mouse_low_level_handler)
+            self.mouse_id = windll.user32.SetWindowsHookExA(self.WH_MOUSE_LL, mouse_pointer,
+                                                            windll.kernel32.GetModuleHandleA(None), 0)
+
+        self.listen()
+
+    def unhook_mouse(self):
+        """Unhook mouse events"""
+        self.mouse_hook = False
+        windll.user32.UnhookWindowsHookEx(self.mouse_id)
+
+    def unhook_keyboard(self):
+        """Unhook keyboard events"""
+        self.keyboard_hook = False
+        windll.user32.UnhookWindowsHookEx(self.keyboard_id)
+
+    def listen(self):
+        """Listen events"""
+        atexit.register(windll.user32.UnhookWindowsHookEx, self.keyboard_id)
+        atexit.register(windll.user32.UnhookWindowsHookEx, self.mouse_id)
+        while self.mouse_hook or self.keyboard_hook:
+            msg = windll.user32.GetMessageW(None, 0, 0, 0)
```

Comment:
In theory we need to wait for every message (`GetMessage` should return if some message is received as far as I understand). We just need to quit on `WM_QUIT`. But I didn't look into `PumpMessages()` implementation yet to know it for sure.